sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.Request.cs

```diff
@@ -32,41 +32,43 @@ protected internal HttpPipelineRequest()
             _headers = new ArrayBackedRequestHeaders();
         }
 
-        protected override string GetMethodCore()
-            => _method;
-
-        protected override void SetMethodCore(string method)
+        protected override string MethodCore
         {
-            Argument.AssertNotNull(method, nameof(method));
+            get => _method;
+            set
+            {
+                Argument.AssertNotNull(value, nameof(value));
 
-            _method = method;
+                _method = value;
+            }
         }
 
-        protected override Uri GetUriCore()
+        protected override Uri UriCore
         {
-            if (_uri is null)
+            get
             {
-                throw new InvalidOperationException("Uri has not been set on this instance.");
+                if (_uri is null)
```

Comment:
We should not have properties that throw unless we have a tester. We need to make properties either required (so they are never null) or nullable.

Reply:
Let me look at whether we can make this required and still have the Azure.Core integration work - i.e. preserve back-compat of the Request semantics.  If not, we can consider making this nullable as you suggested offline

Reply:
This would have large API ripples if we made Uri required, e.g. on CreateMessage on transport and pipeline as well.  I think the best thing at this point is to make it nullable.